src/components/node-list/node-list-tree.js

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import TreeView from '@material-ui/lab/TreeView';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+
+import {
+  toggleModularPipelineActive,
+  toggleModularPipelineFilter,
+} from '../../actions/modular-pipelines';
+import { toggleTypeDisabled } from '../../actions/node-type';
+import { getNodeTypes, getNodeTypeIDs } from '../../selectors/node-types';
+import {
+  getModularPipelineIDs,
+  getModularPipelineData,
+} from '../../selectors/modular-pipelines';
+import {
+  getNestedModularPipelines,
+  getTreeSearchValueCount,
+} from './node-list-items';
+import {
+  getGroupedNodes,
+  getNodeSelected,
+  getNodeModularPipelines,
+} from '../../selectors/nodes';
+import { loadNodeData } from '../../actions/nodes';
+import NodeListTreeItem from './node-list-tree-item';
+
+const useStyles = makeStyles({
+  root: {
+    height: 110,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+});
+
+const StyledTreeView = withStyles({
+  root: {
+    padding: '0 0 0 20px',
+  },
+})(TreeView);
+
+const isModularPipelineType = (type) => type === 'modularPipeline';
+
+const TreeListProvider = ({
+  nodes,
+  nodeSelected,
+  onToggleNodeSelected,
+  searchValue,
+  modularPipelines,
+  modularPipelineIds,
+  nodeModularPipelines,
+  onItemChange,
+  onItemMouseEnter,
+  onItemMouseLeave,
+  nodeTypeIDs,
+}) => {
+  const classes = useStyles();
+
+  const treeData = getNestedModularPipelines({
+    nodes,
+    modularPipelines,
+    nodeSelected,
+    searchValue,
+    modularPipelineIds,
+    nodeModularPipelines,
+    nodeTypeIDs,
+  });
+
+  const searchTreeCount = getTreeSearchValueCount({ treeData, searchValue });
```

Comment:
yes this is a selector I wrote for keeping count for the search results that will be shown on collapse the tree - I have kept this in for the potential next iteration, but since we are not including in this release I will take this out.